modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java

```diff
@@ -1394,7 +1394,7 @@ public CodegenModel fromModel(String name, Model model, Map<String, Model> allDe
                     }
                     // set first interface with discriminator found as parent
                     if (parent == null
-                            && (supportsModelExtension
+                    		&& ((supportsModelExtension && (interfaceModel instanceof ModelImpl || interfaceModel instanceof ComposedModel))
```

Comment:
Added validation of parent model similar to the existing one for discriminator case.